src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,43 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.BulldozerProducer;
+import core.mate.academy.model.ExcavatorProducer;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.TruckProducer;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<? extends Machine> machines;
+        switch (type.getSimpleName()) {
+            case "Excavator" :
+                machines = new ExcavatorProducer().get();
+                break;
+            case "Truck" :
+                machines = new TruckProducer().get();
+                break;
+            case "Bulldozer" :
+                machines = new BulldozerProducer().get();
+                break;
```

Comment:
```suggestion
                return new ArrayList<>(bulldozerProducer.get());
```
new BulldozerProducer() це краще оголосити як поле класу щоб не перегенрювати об'єкт кожного разу